security/access_control.rst

```diff
@@ -349,16 +354,20 @@ access those URLs via a specific port. This could be useful for example for
             xsi:schemaLocation="http://symfony.com/schema/dic/services
                 https://symfony.com/schema/dic/services/services-1.0.xsd">
 
-            <rule path="^/cart/checkout"
-                role="IS_AUTHENTICATED_ANONYMOUSLY"
-                port="8080"
-            />
+            <config>
```

Comment:
not sure we should add `access_control` node here, too

Reply:
Yes, it looks like something is missing ... the example is not exactly the same as the PHP version.

Reply:
I double checked and it is ok, see the output from `config:dump-reference` command:
<details>
www-data@97adf7fdeb83:~$ bin/console config:dump-reference security --format=xml
<!-- Default configuration for extension with alias: "security" -->
<!-- Namespace: http://symfony.com/schema/dic/security -->
<!-- access-denied-url: Example: /foo/error403 -->
<!-- session-fixation-strategy: One of "none"; "migrate"; "invalidate" -->
<config
    access-denied-url="null"
    session-fixation-strategy="migrate"
    hide-user-not-found="true"
    always-authenticate-before-granting="false"
    erase-credentials="true"
>

    <!-- strategy: One of "affirmative"; "consensus"; "unanimous" -->
    <access-decision-manager
        strategy=""
        service=""
        allow-if-all-abstain="false"
        allow-if-equal-granted-denied="true"
    />

    <!-- prototype -->
    <!-- hash-algorithm: Name of hashing algorithm for PBKDF2 (i.e. sha256, sha512, etc..) See hash_algos() for a list of supported algorithms. -->
    <encoder
        class="encoder class"
        algorithm=""
        hash-algorithm="sha512"
        key-length="40"
        ignore-case="false"
        encode-as-base64="true"
        iterations="5000"
        cost="13"
        memory-cost="null"
        time-cost="null"
        threads="null"
        id=""
    />

    <!-- prototype -->
    <provider
        name="provider name"
        id=""
    >

        <chain>

            <!-- prototype -->
            <provider>scalar value</provider>

        </chain>

        <memory>

            <!-- prototype -->
            <user
                name="user name"
                password="null"
            >

                <!-- prototype -->
                <roles>scalar value</roles>

            </user>

        </memory>

        <!-- service: Required -->
        <!-- base-dn: Required -->
        <ldap
            service="ldap"
            base-dn=""
            search-dn=""
            search-password=""
            uid-key="sAMAccountName"
            filter="({uid_key}={username})"
            password-attribute="null"
        >

            <!-- prototype -->
            <default-roles>scalar value</default-roles>

        </ldap>

        <!-- class: Required -->
        <entity
            class=""
            property="null"
            manager-name="null"
        />

        <lexik-jwt class="Lexik\Bundle\JWTAuthenticationBundle\Security\User\JWTUser" />

    </provider>

    <!-- prototype -->
    <!-- user-checker: The UserChecker to use when authenticating users in this firewall. -->
    <!--
        logout-on-user-change: When true, it will trigger a logout for the user if something has changed. Note: No-Op option since 4.0. Will always be true.;
                               Deprecated (The "security.firewalls.logout_on_user_change" configuration key has been deprecated in Symfony 4.1.)
    -->
    <firewall
        name="firewall name"
        pattern=""
        host=""
        security="true"
        user-checker="security.user_checker"
        request-matcher=""
        access-denied-url=""
        access-denied-handler=""
        entry-point=""
        provider=""
        stateless="false"
        context=""
        logout-on-user-change="true"
    >

        <!-- prototype -->
        <methods>scalar value</methods>

        <logout
            csrf-parameter="_csrf_token"
            csrf-token-generator=""
            csrf-token-id="logout"
            path="/logout"
            target="/"
            success-handler=""
            invalidate-session="true"
        >

            <!-- prototype -->
            <delete-cookie
                name="delete cookie name"
                path="null"
                domain="null"
            />

            <!-- prototype -->
            <handler>scalar value</handler>

        </logout>

        <anonymous secret="null" />

        <!-- stateless: Deprecated (The "security.firewalls..switch_user.stateless" configuration key has been deprecated in Symfony 4.1.) -->
        <switch-user
            provider=""
            parameter="_switch_user"
            role="ROLE_ALLOWED_TO_SWITCH"
            stateless="false"
        />

        <x509
            provider=""
            user="SSL_CLIENT_S_DN_Email"
            credentials="SSL_CLIENT_S_DN"
        />

        <remote-user
            provider=""
            user="REMOTE_USER"
        />

        <simple-preauth
            provider=""
            authenticator=""
        />

        <!-- provider: A key from the "providers" section of your security config, in case your user provider is different than the firewall -->
        <!-- entry-point: A service id (of one of your authenticators) whose start() method should be called when an anonymous user hits a page that requires authentication -->
        <guard
            provider=""
            entry-point="null"
        >

            <!-- prototype -->
            <!-- An array of service ids for all of your "authenticators" -->
            <authenticator>scalar value</authenticator>

        </guard>

        <lexik-jwt
            throw-exceptions="false"
            create-entry-point="true"
            authentication-provider="lexik_jwt_authentication.security.authentication.provider"
            authentication-listener="lexik_jwt_authentication.security.authentication.listener"
        >

            <authorization-header
                enabled="true"
                prefix="Bearer"
                name="Authorization"
            />

            <cookie
                enabled="false"
                name="BEARER"
            />

            <query-parameter
                enabled="false"
                name="bearer"
            />

        </lexik-jwt>

        <form-login
            provider=""
            remember-me="true"
            success-handler=""
            failure-handler=""
            check-path="/login_check"
            use-forward="false"
            require-previous-session="false"
            username-parameter="_username"
            password-parameter="_password"
            csrf-parameter="_csrf_token"
            csrf-token-id="authenticate"
            post-only="true"
            always-use-default-target-path="false"
            default-target-path="/"
            login-path="/login"
            target-path-parameter="_target_path"
            use-referer="false"
            failure-path="null"
            failure-forward="false"
            failure-path-parameter="_failure_path"
            csrf-token-generator=""
        />

        <form-login-ldap
            provider=""
            remember-me="true"
            success-handler=""
            failure-handler=""
            check-path="/login_check"
            use-forward="false"
            require-previous-session="false"
            username-parameter="_username"
            password-parameter="_password"
            csrf-parameter="_csrf_token"
            csrf-token-id="authenticate"
            post-only="true"
            always-use-default-target-path="false"
            default-target-path="/"
            login-path="/login"
            target-path-parameter="_target_path"
            use-referer="false"
            failure-path="null"
            failure-forward="false"
            failure-path-parameter="_failure_path"
            csrf-token-generator=""
            service="ldap"
            dn-string="{username}"
            query-string=""
        />

        <json-login
            provider=""
            remember-me="true"
            success-handler=""
            failure-handler=""
            check-path="/login_check"
            use-forward="false"
            require-previous-session="false"
            username-path="username"
            password-path="password"
        />

        <json-login-ldap
            provider=""
            remember-me="true"
            success-handler=""
            failure-handler=""
            check-path="/login_check"
            use-forward="false"
            require-previous-session="false"
            username-path="username"
            password-path="password"
            service="ldap"
            dn-string="{username}"
            query-string=""
        />

        <simple-form
            provider=""
            remember-me="true"
            success-handler=""
            failure-handler=""
            check-path="/login_check"
            use-forward="false"
            require-previous-session="false"
            username-parameter="_username"
            password-parameter="_password"
            csrf-parameter="_csrf_token"
            csrf-token-id="authenticate"
            post-only="true"
            authenticator=""
            always-use-default-target-path="false"
            default-target-path="/"
            login-path="/login"
            target-path-parameter="_target_path"
            use-referer="false"
            failure-path="null"
            failure-forward="false"
            failure-path-parameter="_failure_path"
            csrf-token-generator=""
        />

        <http-basic
            provider=""
            realm="Secured Area"
        />

        <http-basic-ldap
            provider=""
            realm="Secured Area"
            service="ldap"
            dn-string="{username}"
            query-string=""
        />

        <!-- login-path: Required -->
        <oauth
            provider=""
            remember-me="true"
            success-handler=""
            failure-handler=""
            check-path="/login_check"
            use-forward="false"
            require-previous-session="false"
            always-use-default-target-path="false"
            default-target-path="/"
            login-path=""
            target-path-parameter="_target_path"
            use-referer="false"
            failure-path="null"
            failure-forward="false"
            failure-path-parameter="_failure_path"
        >

            <oauth-user-provider
                service=""
                oauth=""
            >

                <!-- class: Required -->
                <orm
                    class=""
                    manager-name="null"
                >

                    <!-- prototype -->
                    <properties name="properties name">scalar value</properties>

                </orm>

                <fosub>

                    <!-- prototype -->
                    <properties name="properties name">scalar value</properties>

                </fosub>

            </oauth-user-provider>

            <!-- prototype -->
            <resource-owners name="resource owners name">scalar value</resource-owners>

        </oauth>

        <!-- secret: Required -->
        <!-- secure: One of true; false; "auto" -->
        <!-- samesite: One of null; "lax"; "strict" -->
        <remember-me
            secret=""
            token-provider=""
            catch-exceptions="true"
            name="REMEMBERME"
            lifetime="31536000"
            path="/"
            domain="null"
            secure="false"
            httponly="true"
            samesite="null"
            always-remember-me="false"
            remember-me-parameter="_remember_me"
        >

            <!-- prototype -->
            <user-provider>scalar value</user-provider>

        </remember-me>

    </firewall>

    <!-- prototype -->
    <!--
        path: use the urldecoded format;
              Example: ^/path to resource/
    -->
    <rule
        requires-channel="null"
        path="null"
        host="null"
        port="null"
        allow-if="null"
    >

        <!-- prototype -->
        <ip>scalar value</ip>

        <!-- prototype -->
        <method>scalar value</method>

        <!-- prototype -->
        <role>scalar value</role>

    </rule>

    <!-- prototype -->
    <role id="role id">

        <!-- prototype -->
        <id>scalar value</id>

    </role>

</config>
</details>

Reply:
Fyi: `rule` is configured to be the singular version of `access_control` (like `path` is the singular element of a `paths` list in Yaml).